imodels/rule_set/boosted_rules.py

```diff
@@ -18,9 +19,13 @@ class BoostedRulesClassifier(RuleSet, BaseEstimator, MetaEstimatorMixin, Classif
     Currently limited to only binary classification.
     '''
 
-    def __init__(self, n_estimators=10, estimator=partial(DecisionTreeClassifier, max_depth=1)):
+    def __init__(self, method='DT', n_estimators=10):
```

Comment:
Not sure what a good design for handling depreciation of estimator as partial instead of string here. This approach will cause problems with existing code on user end.

Reply:
Agreed, it also makes it difficult to pass arguments (e.g. max_depth) to the estimator if it only takes a string. I think this argument should be preserved the way it is, so it doesn't break backward compatibility. In addition, if the argument to estimator is of type string, then check if it's "SLIPPER" and then set it the way you have used here.

Reply:
Do you think it's worth spending some time to wrap this closer to the procedure used in sklearn boosting algorithms? For example have the BoostedRulesClassifier inherit from [BaseWeightBoosting](https://github.com/scikit-learn/scikit-learn/blob/2beed55847ee70d363bdbfe14ee4401438fba057/sklearn/ensemble/_weight_boosting.py#L52) then extend the validate and fit functions and implement the _boost function for the BoostedRulesClassifier?

Reply:
I think that might be overkill for what we have now (unless we implement many more boosting-type algorithms in the near future) - for now it's probably easiest to just have estimator be a flexible argument.

Reply:
Yeah if there isn't any intention to immediately expand the flexibility of the boosting procedure then I would agree. I updated to accept the partial estimator and then check that object type once instantiated instead of the string